File: Java-Core/src/main/java/presto/type/BooleanType.java
package presto.type;

import presto.error.PrestoError;
import presto.runtime.Context;
import presto.value.Boolean;
import presto.value.ICollection;
import presto.value.IValue;
import presto.value.ListValue;

public class BooleanType extends NativeType {
	
	static BooleanType instance = new BooleanType();
	
	public static BooleanType instance() {
		return instance;
	}
	
	private BooleanType() {
		super("Boolean");
	}

	@Override
	public Class<?> toJavaClass() {
		return java.lang.Boolean.class;
	}
	
	@Override
	public boolean isAssignableTo(Context context, IType other) {
		return (other instanceof BooleanType) || (other instanceof AnyType);
	}
	
	@Override
	public ListValue sort(Context context, ICollection<IValue> list) throws PrestoError {
		return this.doSort(context,list);
	}
	
	@Override
	public IValue convertJavaValueToPrestoValue(Object value) {
        if (value instanceof java.lang.Boolean)
            return Boolean.ValueOf(((java.lang.Boolean)value).booleanValue());
        else
            return (IValue)value; // TODO for now
	}

}

File: Java-CodeServer/src/main/resources/js/prompto/type/BooleanType.js
var NativeType = require("./NativeType").NativeType;
var AnyType = require("./AnyType").AnyType;
var Bool = null;

exports.resolve = function() {
    Bool = require("../value/Bool").Bool;
}

function BooleanType()  {
	NativeType.call(this, "Boolean");
	return this;
}

BooleanType.prototype = Object.create(NativeType.prototype);
BooleanType.prototype.constructor = BooleanType;

BooleanType.instance = new BooleanType();

/*
@Override
public Class<?> toJavaClass() {
	return java.lang.Boolean.class;
}
*/

BooleanType.prototype.isAssignableTo = function(context, other) {
	return (other instanceof BooleanType) || (other instanceof AnyType);
};

BooleanType.prototype.checkAnd = function(context, other) {
	if(other instanceof BooleanType) {
		return BooleanType.instance;
	} else {
		return NativeType.prototype.checkAnd.call(this, context, other);
	}
};

BooleanType.prototype.checkOr = function(context, other) {
	if(other instanceof BooleanType) {
		return BooleanType.instance;
	} else {
		return NativeType.prototype.checkOr.call(this, context, other);
	}
};

BooleanType.prototype.checkNot = function(context) {
	return BooleanType.instance;
};


BooleanType.prototype.convertJavaScriptValueToPrestoValue = function(context, value, returnType) {
	if (typeof(value)=='boolean') {
		return Bool.ValueOf(value);
	} else {
		return value; // TODO for now
	}
};

exports.BooleanType = BooleanType;